checkQC/run_type_recognizer.py

```diff
@@ -101,16 +101,36 @@ def name():
     @staticmethod
     def reagent_version(runtype_recognizer):
         """
-        Find the reagent version used for this run, as MiSeqs can have multiple
-        different reagent kit versions.
-
-        :returns: reagent version of format v[number] e.g. v3
-        """
-        try:
-            reagent_version = runtype_recognizer.run_parameters["RunParameters"]["ReagentKitVersion"]
-            return reagent_version.replace("Version", "v")
-        except KeyError:
-            raise ReagentVersionUnknown("No reagent version specified for this instrument type")
+        Find the reagent kit version (and flowcell mode if applicable) for this run
+
+        :returns: reagent version in format v[reagent kit version] or [flowcell mode]_v[reagent kit version],
+         e.g. v3 or nano_v2
+        """
+        def _reagent_kit_version(runtype_recognizer):
+            try:
+                reagent_version = runtype_recognizer.run_parameters["RunParameters"]["ReagentKitVersion"]
+                return reagent_version.replace("Version", "v")
+            except KeyError:
+                raise ReagentVersionUnknown("No reagent version specified for this instrument type")
+
+        def _flowcell_type(runtype_recognizer):
+            try:
+                tiles_per_swath = int(runtype_recognizer.run_parameters["RunParameters"]["Setup"]["NumTilesPerSwath"])
+                if tiles_per_swath == 2:
+                    return "nano"
+                elif tiles_per_swath == 4:
+                    return "micro"
+                elif tiles_per_swath >= 14:
+                    return "standard"
```

Comment:
Is it possible that `tiles_per_swath` does not match any of these values? In this case maybe you could raise the `ReagentVersionUnknown` exception?

Reply:
Ah good catch, I think it shouldn't be possible with the current MiSeq configurations, but Illumina might add something new in the future. Will fix!